benchmarks/benchmarks/generate_data.py

```diff
@@ -0,0 +1,187 @@
+"""
+Scripts for generating supporting data for benchmarking.
+
+Data generated using iris-esmf-regrid should use
+:func:`run_function_elsewhere`, which means that data is generated using a
+fixed version of iris-esmf-regrid and a fixed environment, rather than those
+that get changed when the benchmarking run checks out a new commit.
+
+Downstream use of data generated 'elsewhere' requires saving; usually in a
+NetCDF file. Could also use pickling but there is a potential risk if the
+benchmark sequence runs over two different Python versions.
+
+"""
+from inspect import getsource
+from subprocess import CalledProcessError, check_output, run
+from os import environ
+from pathlib import Path
+import re
+from textwrap import dedent
+
+from iris import load_cube
+from iris.experimental.ugrid import PARSE_UGRID_ON_LOAD
+
+#: Python executable used by :func:`run_function_elsewhere`, set via env
+#:  variable of same name. Must be path of Python within an environment that
+#:  supports iris-esmf-regrid and has iris-esmf-regrid installed via
+#:  ``pip install -e``.
+try:
+    DATA_GEN_PYTHON = environ["DATA_GEN_PYTHON"]
+    _ = check_output([DATA_GEN_PYTHON, "-c", "a = True"])
+except KeyError:
+    error = "Env variable DATA_GEN_PYTHON not defined."
+    raise KeyError(error)
+except (CalledProcessError, FileNotFoundError, PermissionError):
+    error = "Env variable DATA_GEN_PYTHON not a runnable python executable path."
+    raise ValueError(error)
+
+default_data_dir = (Path(__file__).parent.parent / ".data").resolve()
+BENCHMARK_DATA = Path(environ.get("BENCHMARK_DATA", default_data_dir))
+if BENCHMARK_DATA == default_data_dir:
+    BENCHMARK_DATA.mkdir(exist_ok=True)
+elif not BENCHMARK_DATA.is_dir():
+    message = f"Not a directory: {BENCHMARK_DATA} ."
+    raise ValueError(message)
+
+# Flag to allow the rebuilding of synthetic data.
+REUSE_DATA = True
```

Comment:
My thinking is that this would work the same way that CACHE_BUILD works in cirrus.yml where it is changed in a commit when files need to be refreshed. This allows cached files on the GitHub CI to be refreshed by a commit in a way which is clear and readable.